Sources/Private/CoreAnimation/ValueProviderStore.swift

```diff
@@ -40,6 +40,11 @@ final class ValueProviderStore {
     valueProviders.append((keypath: keypath, valueProvider: valueProvider))
   }
 
+  /// Removes all ValueProviders for the given `AnimationKeypath`
+  func removeValueProvider(keypath: AnimationKeypath) {
+      valueProviders.removeAll(where: { $0.keypath == keypath })
```

Comment:
What about if the given `keypath` is something like `Layer.*`? Should it remove key paths like `Layer.Color`? It seems like that probably wouldn't happen with the current implementation using `==`.

Reply:
Could you please add a test case for this to `ValueProvidersTests.swift`, with some cases like that where the the key path being removed isn't directly present in the list of providers and instead matches using wildcards?

Reply:
Is that behavior really expected? The method is named `removeValueProvider`, which suggests that it should only remove the exact value provider for the specified keyPath, and not multiple `ValueProviders`.
Another point - in `setValueProvider` the value providers are removed the same way as is implemented in this method (see here - https://github.com/airbnb/lottie-ios/blob/master/Sources/Private/CoreAnimation/ValueProviderStore.swift#L39). If you anyway would like to have functionality for wildcard case (like `Layer.*`) - I'm not sure how exactly it should work, so help would be wanted.
I added test for the current `removeValueProvider` implementation. 

Thank you

Reply:
> Is that behavior really expected?

Good question, I can see it making sense for the behavior to stay as-is in the Core Animation rendering engine.

However, I'm pretty sure that this behavior that I describe is the current behavior implemented in this PR for the Main Thread rendering engine. We should make sure the two implementations are consistent.